src/display/canvas.js

```diff
@@ -922,6 +922,11 @@ const CanvasGraphics = (function CanvasGraphicsClosure() {
       }
       this.ctx.transform.apply(this.ctx, viewport.transform);
 
+      this.viewportTransform = viewport.transform;
+      // Square of viewport scale factor.
+      this.viewportSqScaleFactor =
+        viewport.transform[0] ** 2 + viewport.transform[2] ** 2;
+      this.viewportScaleFactor = Math.sqrt(this.viewportSqScaleFactor);
```

Comment:
This doesn't actually take the optional `transform` argument into account, so I'd expect that you actually want to use `this.baseTransform` in your calculations here?

Furthermore, you're not actually using most of these properties anywhere but here, hence please use local variables where appropriate!

It should be possible to simplify all of this code, after moving it down a couple of lines, to e.g. the following:
```suggestion
      this._combinedScaleFactor =
        Math.sqrt(this.baseTransform[0] ** 2 + this.baseTransform[2] ** 2);
```

Reply:
I forgot to remove them.
Anyway, `baseTransform` is likely better than `viewport.transform`: thanks for the tip.